src/clfortran.f90

```diff
@@ -426,39 +426,16 @@ module clfortran
     integer(c_int32_t), parameter :: CL_PROFILING_COMMAND_START                 = Z'1282'
     integer(c_int32_t), parameter :: CL_PROFILING_COMMAND_END                   = Z'1283'
     
-    ! ------------
-    ! Types
-    ! ------------
-    
-    type, BIND(C) :: cl_image_format 
-        integer(c_int32_t) :: image_channel_order
-        integer(c_int32_t) :: image_channel_data_type
-    end type
-    
-    type, BIND(C) :: cl_image_desc
-        integer(c_int32_t)  :: image_type
-        integer(c_size_t)   :: image_width
-        integer(c_size_t)   :: image_height
-        integer(c_size_t)   :: image_depth
-        integer(c_size_t)   :: image_array_size
-        integer(c_size_t)   :: image_row_pitch
-        integer(c_size_t)   :: image_slice_pitch
-        integer(c_int32_t)  :: num_mip_levels
-        integer(c_int32_t)  :: num_samples
-        integer(c_intptr_t) :: buffer
-    end type
-    
-
     !
     ! Start interfaces.
     !
-    contains
 
     ! ------------
     ! Platform API
     ! ------------
 
     ! clGetPlatformIDs
+    interface
```

Comment:
I think this lack of the use of `interface` must be the reason I had terrible trouble originally when I included clfortran.o in the objects being compiled - I got a binary but it did nothing.